sbin/build.sh

```diff
@@ -684,38 +684,18 @@ createOpenJDKFailureLogsArchive() {
     createArchive "${adoptLogArchiveDir}" "${makeFailureLogsName}"
 }
 
-# Setup JAVA env to run "ant task"
+# Setup JAVA env to run "ant task" for CycloneDX build
 setupAntEnv() {
   local javaHome=""
 
-  if [ ${JAVA_HOME+x} ] && [ -d "${JAVA_HOME}" ]; then
-    javaHome=${JAVA_HOME}
-  elif [ ${JDK8_BOOT_DIR+x} ] && [ -d "${JDK8_BOOT_DIR}" ]; then
-    javaHome=${JDK8_BOOT_DIR}
-  elif [ ${JDK11_BOOT_DIR+x} ] && [ -d "${JDK11_BOOT_DIR}" ]; then
-    javaHome=${JDK11_BOOT_DIR}
-  elif [ ${BUILD_CONFIG[JDK_BOOT_DIR]+x} ] && [ -d "${BUILD_CONFIG[JDK_BOOT_DIR]}" ]; then
-  # fall back to use JDK_BOOT_DIR which is set in make-adopt-build-farm.sh
-    javaHome="${BUILD_CONFIG[JDK_BOOT_DIR]}"
-  else
-    echo "Unable to find a suitable JAVA_HOME to build the cyclonedx-lib"
-    exit 2
-  fi
-  echo "${javaHome}"
-}
+  # TemurinGenSBOM.java requires jdk-17+ to build openkeystore
+  apiURL="https://api.adoptium.net/v3/binary/latest/17/ga/${OPERATING_SYSTEM}/${BUILD_CONFIG[OS_ARCHITECTURE]}/jdk/hotspot/normal/eclipse"
```

Comment:
Do we really need to do this? Most of our build machines have java 17 installed in a known location so downloading it every time seems a bit over the top.

Reply:
Solaris is a valid point.

Reply:
@sxa What environment would point at it?
Also, the above is generic for user of this script

Reply:
Yeah thought about Solaris, not sure what we do there....yet

Reply:
I'd like to just get some thing building rather than none :-)

Reply:
> Also, the above is generic for user of this script
Yes but "if it's not in our location, do the download" would seem a sensible thing to do (and the fewer things we download during the process the better and more secure we will be.

Reply:
So openkeystore is not going to build on Solaris, so I think we're going to have to re-factor that code out of TemurinGenSBOM and into another signing app, called in a standalone job

Reply:
@sxa in fact i'm thinking refactor the whole class into a separate invoked build-farm job, a bit like GPG signing. Trying to perform at the end of the build is troublsome